org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java

```diff
@@ -0,0 +1,117 @@
+package org.eclipse.lsp4xml.extensions.maven;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.Repository;
+import org.apache.maven.plugin.InvalidPluginDescriptorException;
+import org.apache.maven.plugin.MavenPluginManager;
+import org.apache.maven.plugin.PluginDescriptorParsingException;
+import org.apache.maven.plugin.PluginResolutionException;
+import org.apache.maven.plugin.descriptor.MojoDescriptor;
+import org.apache.maven.plugin.descriptor.Parameter;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.repository.RemoteRepository.Builder;
+import org.eclipse.lsp4j.MarkupContent;
+import org.eclipse.lsp4xml.dom.DOMNode;
+import org.eclipse.lsp4xml.services.extensions.IPositionRequest;
+
+public class Utils {
```

Comment:
We need a clearer name than Utils or this class would become too messy.
I think we should split it into 2 parts `MavenPluginUtils` for the Maven parts to reuse, and `DOMUtils` for the plain DOM parts. In fine, it's likely that we try to replace the `DOMUtils` by some API provided in LemMinX directly if existing, or that we migrate our DOMUtils to LemMinX.

Reply:
Agreed, maybe we should show `DomUtils` to the LeMiNX upstream sometime in the near future.